src/Run/Run.tsx

```diff
@@ -86,32 +83,77 @@ class SpielServerRequest {
 
 }
 
+let code = "";
+let filename = "";
+
 const Run = (props) => {
-    let [input,setInput] = React.useState('');
 
-    function run() {
-        props.commands.push(input);
-        props.runCode();
-        setInput("");
+    let [gameHistory, setGameHistory] = React.useState(Array<string>());
+
+    code = props.code;
```

Comment:
Ohhh...maybe this line fixed it?

Reply:
Yeah I think this is the line that fixed it! Thanks :)